plugins/MCWhitelist/__init__.py

```diff
@@ -0,0 +1,192 @@
+import discord
+from discord.ext import commands
+
+from Plugin import AutomataPlugin
+from Globals import (
+    mongo_client,
+    PRIMARY_GUILD,
+    VERIFIED_ROLE,
+    EXECUTIVE_ROLE,
+)
+from .mojang_api import MojangAPI
+from .whitelist_http_api import WhitelistHttpApi
+
+
+def is_executive():
+    async def predicate(ctx):
+        return len([role for role in ctx.author.roles if role.id == EXECUTIVE_ROLE]) > 0
+
+    return commands.check(predicate)
+
+
+class MCWhitelist(AutomataPlugin):
+    """Provides Minecraft account whitelisting for MUNCS Craft."""
+
+    def __init__(self, manifest, bot: commands.Bot):
+        super().__init__(manifest, bot)
+
+        self.whitelisted_accounts = (
+            mongo_client.automata.mcwhitelist_whitelisted_accounts
+        )
+        self.disallowed_members = mongo_client.automata.mcwhitelist_disallowed_members
+        self.mojang_api = MojangAPI()
+        self.whitelist_http_api = WhitelistHttpApi()
+
+    async def get_whitelisted_account(self, member):
+        query = {"discord_id": member.id}
+        return await self.whitelisted_accounts.find_one(query)
+
+    async def is_minecraft_account_already_associated(self, username):
+        whitelist = self.whitelist_http_api.whitelist()
+
+        for entry in whitelist:
+            if entry["name"] == username:
+                return True
+
+        return False
+
+    async def is_disallowed(self, member):
+        query = {"discord_id": member.id}
+        return await self.disallowed_members.find_one(query) != None
+
+    async def remove_whitelisted_account(self, ctx, whitelisted_account):
+        username = whitelisted_account["minecraft_username"]
+        self.whitelist_http_api.remove(username)
+        await self.whitelisted_accounts.delete_many({"discord_id": ctx.author.id})
+
+    async def account_embed(self, whitelisted_account):
+        embed = discord.Embed()
+
+        profile = await self.mojang_api.profile_from_uuid(
+            whitelisted_account["minecraft_uuid"]
+        )
+
+        if profile:
+            skin_url = self.mojang_api.skin_url_from_profile(profile)
+            if skin_url:
```

Comment:
Same here, rather than checking if the value is truthy, you should check if the value is not None